js/script.js

```diff
@@ -0,0 +1,87 @@
+// Интерактивная карта по клику
+
+var maplink = document.querySelector(".contacts-and-map__image");
+
+if (maplink) {
+  var mappopup = document.querySelector(".popup-map");
+  var mapclose = mappopup.querySelector(".form-close-button");
+
+  maplink.addEventListener("click", function(evt) {
+    evt.preventDefault();
+    mappopup.classList.add("modal-show");
+  });
+
+  mapclose.addEventListener("click", function(evt) {
+    evt.preventDefault();
+    mappopup.classList.remove("modal-show");
+  });
+
+  window.addEventListener("keydown", function(evt) {
+    if (evt.keyCode === 27) {
+      evt.preventDefault();
+      if (mappopup.classList.contains("modal-show")) {
+        mappopup.classList.remove("modal-show");
+      }
+    }
+  });
+}
+
+// Корзина в каталоге
+var buyButton = document.querySelectorAll(".catalog-goods-list__buy-button");
+var cart = document.querySelector(".cart");
+var buyClose = cart.querySelector(".form-close-button");
+var buyContinue = cart.querySelector(".cart__button-continue")
+
+buyButton.forEach(function(item) {
+  item.addEventListener("click", function(evt) {
+    evt.preventDefault();
+    cart.classList.add("modal-show");
+  });
+});
+
+buyClose.addEventListener("click", function(evt) {
+  evt.preventDefault();
+  cart.classList.remove("modal-show");
+});
+
+window.addEventListener("keydown", function(evt) {
+  if (evt.keyCode === 27) {
+    evt.preventDefault();
+    if (cart.classList.contains("modal-show")) {
+      cart.classList.remove("modal-show");
+    }
+  }
+});
+
+// Красная корзина
+var cartButton = document.querySelector(".site-navigation-list__item:nth-child(2)");
+var bookmarksButton = document.querySelector(".catalog-goods-list__bookmarks-button");
+if (bookmarksButton) {
+  var bookmarksButton = document.querySelector(".catalog-goods-list__bookmarks-button");
```

Comment:
```suggestion
  bookmarksButton = document.querySelector(".catalog-goods-list__bookmarks-button");
```

ты уже объявил эту переменную на 58 строке, теперь ей можно пользоваться